src/consumer/consumerGroup.js

```diff
@@ -119,6 +120,21 @@ module.exports = class ConsumerGroup {
     this.offsetManager.resolveOffset({ topic, partition, offset })
   }
 
+  /**
+   * Update the consumer offset for the given topic/partition. This will be used
+   * on the next fetch. If this API is invoked for the same topic/partition more
+   * than once, the latest offset will be used on the next fetch.
+   *
+   * @param {string} topic
+   * @param {number} partition
+   * @param {string} offset
+   */
+  seek({ topic, partition, offset }) {
```

Comment:
Given that you only want a single offset per topic and partition, I would suggest changing this to using a Map instead, keying by topic and partition.

```
seek({ topic, partition, offset }) {
  seekOffset.set([topic, partition], offset)
}
```

The usage of the seekOffset becomes a little more tedious, as you have to get an iterable, get the first element and then delete it from the map, but I think it's worth it because you would be guaranteed uniqueness per topic/partition without this somewhat confusing code (took me longer than I'd like to admit to understand why you were filtering the offsets).

To mitigate the annoyance of not having `pop` for a map, you could wrap it into another datastructure that does that under the hood.

Reply:
Fixed this in b276ccf. Let me know if you think it's alright.